src/modules/claim/services/claim-verification.service.ts

```diff
@@ -0,0 +1,193 @@
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc")
+   * @return Boolean indicating if verified and array of error messages
+   */
+  public async verifyRoleEIP191JWT(
+    roleEIP191JWT: RoleEIP191JWT,
+    subjectDID: string,
+    roleNamespace: string
+  ): Promise<{ isVerified: boolean; errors: string[] }> {
+    const { payload, eip191Jwt } = roleEIP191JWT;
+    const errors: string[] = [];
+    const issuerDID = roleEIP191JWT.payload?.iss;
+    if (!issuerDID) {
+      throw new Error('No issuer specified for credential');
+    }
+    const proofVerified = await this.verifyPublicClaim(
+      eip191Jwt,
+      payload?.iss as string
+    );
+    if (!proofVerified) {
+      errors.push(
+        `Verification failed for ${roleNamespace} for ${subjectDID}: Proof not verified for role`
+      );
+    }
+    // Date.now() and JWT expiration time both identify the time elapsed since January 1, 1970 00:00:00 UTC
+    const isExpired = payload?.exp && payload?.exp * 1000 < Date.now();
+    if (isExpired) {
+      errors.push(
+        `Verification failed for ${roleNamespace} for ${subjectDID}: Credential for prerequisite role expired`
+      );
+    }
+    const { verified: issuerVerified, error } =
+      await this.issuerVerificationService.verifyIssuer(
+        issuerDID,
+        payload?.claimData?.claimType
+      );
+    if (!issuerVerified && error) {
+      throw new Error(
+        `Verification failed for ${roleNamespace} for ${subjectDID}: Issuer verification failed: ${error}`
+      );
+    }
+    return {
+      errors: errors,
+      isVerified: !!proofVerified && issuerVerified && !isExpired,
+    };
+  }
+
+  /**
+   * Verifies issued token of the public claim.
+   *
+   * @param {String} token JWT token of the public claim
+   * @param {String} iss DID of the issuer
+   * @return DID of the authenticated identity on successful verification or null otherwise
+   */
+  public async verifyPublicClaim(
+    token: string,
+    did: string
+  ): Promise<string | null> {
+    const didDoc = await this.didService.getById(did);
+    const verifier = new ProofVerifier(didDoc);
+    return verifier.verifyAssertionProof(token);
+  }
+
+  /**
+   * Verifies that a user's Did Document contains all roles required for enrolment (enrolment preconditions)
+   * @param userDID the Did of the user seeking to obtain the role. Must posses enrolment preconditions
+   * @param conditions enrolment preconditions needed to obtain the role (claimType)
+   */
+  public async verifyClaimPresentInDidDocument({
+    userDID,
+    conditions,
+  }: {
+    userDID: string;
+    conditions: string[];
+  }) {
+    const didDocument = await this.didService.getById(userDID);
+    return didDocument.service.some(
+      ({ claimType }) => claimType && conditions.includes(claimType as string)
+    );
+  }
+
+  public async verifyEnrolmentPreconditions(
```

Comment:
I moved the duplicative logic for checking conditions out of claim and claim-issuance service into this file, so they can share the same logic and not repeat.